### DISCUSSION TITLE
retain variable/ tag

### DISCUSSION THREAD

**Discussion by TomasAdvance** [frangoteam/FUXA#781] [General]
Hi Fuxa team,
Is there any possibility for FUXA to remember the server tag value after restarting FUXA? Something like retain variable in PLC.

**Comment by unocelli** [comment]
Hi,
I know the issue but unfortunately the feature has not yet been implemented.

**Comment by TomasAdvance** [comment]
Thanks for your reply.
It is possibility to run the script if the fuxa was restarted?
I am looking for some event which tell me the fuxa start up. Then I am able to write my own code.
Thanks for your answer.
T

**Reply by unocelli** [reply]
that is exactly what is missing, scripts to be executed at start-up. however, you could use an interval script and use a FUXA server tag as a flag to set as initialized

**Comment by rvbatista** [comment]
This cannot be done at the init of the variable? recovering last data from storage if a setting indicate that it is and retain variable?

**Comment by unocelli** [comment]
Hi, I make some test and I found that could work. you make a script SERVER mode and for example 5 seconds interval:

```
const fs = require('fs');

const restoreFilePath = '_reports/myfile.json';

try {
  var runningFlag = $getTag('t_7b44a4b9-5f324e3d' /* FUXA Server - bool-server */);
  
  console.log('script');
  
  if (runningFlag) {
    // save value on running
    console.log('running');

    // write values in file
    var valueToSave = $getTag('t_da476682-ae144701' /* opcua - Byte */);

    var objectToSave = { value: valueToSave };
    const data = JSON.stringify(objectToSave, null, 2);

    fs.writeFile(restoreFilePath, data, (err) => {
      if (err) throw err;
      console.log('Object has been saved to file!');
    });

  } else {
    // just started
    console.log('sarted');

    if (fs.existsSync(restoreFilePath)) {
    // read values to restore
      fs.readFile(restoreFilePath, (err, data) => {
        if (err) throw err;
        const objectFromJson = JSON.parse(data);
        $setTag('t_da476682-ae144701' /* opcua - Byte */, objectFromJson.value);
        console.log(objectFromJson);
      });
    }

    // restore the values
    runningFlag = true;
    $setTag('t_7b44a4b9-5f324e3d' /* FUXA Server - bool-server */, runningFlag);
  }
} catch (err) {
  console.error(err);
}
```
